File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        # modified neetcode solution
        
        longest = 0
        
        numSet = set(nums)
        
        for num in numSet:
            
            length = 1
            if (num-1) not in numSet:
                while (num+length) in numSet:
                    length += 1
                longest = max(length, longest)
        return longest
    
    
    
                    
            

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // O(n) time and space modified neetcode solution
        int longest = 0;
        unordered_set<int> numSet(nums.begin(), nums.end());
        
        for (auto& num: numSet)
        {
            if(numSet.find(num-1) == numSet.end())
            {
                int length = 1;
                while(numSet.find(num+length) != numSet.end())
                {
                    length++;
                }
                longest = max(longest, length);
            }
        }
        return longest;
    }
};